logger.go

```diff
@@ -31,46 +40,91 @@ func New(config Config) (*MicroLogger, error) {
 		config.IOWriter = DefaultIOWriter
 	}
 
-	kitLogger := kitlog.NewJSONLogger(kitlog.NewSyncWriter(config.IOWriter))
-
-	kitLogger = kitlog.With(
-		kitLogger,
-		"caller", config.Caller,
-		"time", config.TimestampFormatter,
-	)
+	log := log.New(config.IOWriter, "", 0)
 
-	l := &MicroLogger{
-		logger: kitLogger,
+	logger := &MicroLogger{
+		human: config.Human,
+		keyVals: map[interface{}]interface{}{
+			"caller": config.Caller,
+			"time":   config.TimestampFormatter,
+			"level":  "info",
+		},
+		log: log,
 	}
 
-	return l, nil
+	return logger, nil
+}
+
+func ToString(v interface{}) string {
+	valuer, ok := v.(Valuer)
+	if ok {
+		v = valuer()
+	}
+	return fmt.Sprintf("%v", v)
 }
 
 func (l *MicroLogger) Log(keyVals ...interface{}) error {
-	return l.logger.Log(keyVals...)
+	combined := map[string]string{}
+	for k, v := range l.keyVals {
+		k := ToString(k)
+		v := ToString(v)
+		combined[k] = v
+	}
+	for i := 0; i < len(keyVals); i += 2 {
+		k := ToString(keyVals[i])
+		v := ToString(keyVals[i+1])
+		combined[k] = v
+	}
+	var encoded string
+	if l.human {
```

Comment:
Maybe split this up into two functions i.e. `logHuman` and `logJSON`.

A different approach I've seen is to create a `humanLogger` and a `jsonLogger` struct which then implement a `Logger` interface.

Reply:
I think that's a good refactoring. There was a lot of common logic being changed so I intermingled them to start. I'll break them into two structs when I get a chance.